crates/noirc_frontend/src/ast/expression.rs

```diff
@@ -382,8 +382,7 @@ pub enum FunctionReturnType {
 
 /// Describes the types of smart contract functions that are allowed.
 /// - All Noir programs in the non-contract context can be seen as `Secret`.
-#[derive(serde::Serialize, serde::Deserialize, Debug, Copy, Clone, PartialEq, Eq)]
-#[serde(rename_all = "snake_case")]
```

Comment:
This seems like a breaking change, why was this needed?

Reply:
This wasn't used anywhere otherwise it wouldn't compile 🙃

Reply:
I removed it from this PR but will followup to remove serde from the frontend crate.

Reply:
Oh thats suprising haha -- I thought it was being used for contract ABIs 🤔

Reply:
I think it's be translated somewhere else. I'll still do it in a follow up PR since it is unrelated to this code but was a nice cleanup to remove the dep